NPE after resuming from background

This wasn't showing up previously and it's not particularly clear what caused it. Possibly

https://github.com/HackIllinois/android-2017/blob/31b5c0f9186a1cd647975fe8e64057abbd901a57/app/src/main/java/org/hackillinois/android/ui/SplashActivity.java#L147

Here's a log of this
```java
java.lang.RuntimeException: Unable to resume activity {org.hackillinois.androidapp/org.hackillinois.android.ui.MainActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Application android.app.Activity.getApplication()' on a null object reference
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3832)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3872)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3014)
	at android.app.ActivityThread.-wrap14(ActivityThread.java)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1650)
	at android.os.Handler.dispatchMessage(Handler.java:102)
	at android.os.Looper.loop(Looper.java:154)
	at android.app.ActivityThread.main(ActivityThread.java:6688)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1468)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1358)
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'android.app.Application android.app.Activity.getApplication()' on a null object reference
	at org.hackillinois.android.ui.base.BaseFragment.getApi(BaseFragment.java:21)
	at org.hackillinois.android.ui.modules.home.HomeFragment.fetchEvents(HomeFragment.java:113)
	at org.hackillinois.android.ui.modules.home.HomeFragment.sync(HomeFragment.java:106)
	at org.hackillinois.android.ui.MainActivity.onResumeFragments(MainActivity.java:132)
	at android.support.v4.app.FragmentActivity.onPostResume(FragmentActivity.java:471)
	at android.support.v7.app.AppCompatActivity.onPostResume(AppCompatActivity.java:171)
	at android.app.Activity.performResume(Activity.java:7082)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3809)
	... 10 more
```

--- a/app/src/main/java/org/hackillinois/android/ui/SplashActivity.java
+++ b/app/src/main/java/org/hackillinois/android/ui/SplashActivity.java
@@ -8,7 +8,6 @@ import android.view.WindowManager;
 
 import com.annimon.stream.Optional;
 
-import org.hackillinois.android.BuildConfig;
 import org.hackillinois.android.R;
 import org.hackillinois.android.api.response.location.LocationResponse;
 import org.hackillinois.android.api.response.login.LoginResponse;
@@ -129,16 +128,13 @@ public class SplashActivity extends BaseActivity {
 	private void moveOn() {
 		Timber.d("Moving to %s", activityClass.getSimpleName());
 		Intent i = new Intent(this, activityClass);
-		i.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
 		startActivity(i);
 		finish();
 	}
 
 	@OnClick(R.id.splash_anim)
 	public void skipSplash() {
-		if (BuildConfig.DEBUG) {
-			Timber.i("Skipping splash screen");
-			moveOn();
-		}
+		Timber.i("Skipping splash screen");
+		moveOn();
 	}
 }

--- a/app/src/main/java/org/hackillinois/android/ui/base/BaseFragment.java
+++ b/app/src/main/java/org/hackillinois/android/ui/base/BaseFragment.java
@@ -3,21 +3,20 @@ package org.hackillinois.android.ui.base;
 import android.os.Bundle;
 import android.support.annotation.Nullable;
 import android.support.v4.app.Fragment;
-import android.support.v4.view.LayoutInflaterCompat;
-import android.support.v7.app.AppCompatActivity;
-
-import com.mikepenz.iconics.context.IconicsLayoutInflater2;
 
 import org.hackillinois.android.App;
 import org.hackillinois.android.api.HackIllinoisAPI;
 
 public class BaseFragment extends Fragment {
+	private HackIllinoisAPI api;
+
 	@Override
 	public void onCreate(@Nullable Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
+		api = ((App) getActivity().getApplication()).getApi();
 	}
 
 	public HackIllinoisAPI getApi() {
-		return ((App) getActivity().getApplication()).getApi();
+		return api;
 	}
 }

--- a/app/src/main/java/org/hackillinois/android/ui/modules/profile/ProfileFragment.java
+++ b/app/src/main/java/org/hackillinois/android/ui/modules/profile/ProfileFragment.java
@@ -111,8 +111,10 @@ public class ProfileFragment extends BaseFragment {
 		Uri userBadge = Uri.parse(result.getContents());
 		String textId = userBadge.getQueryParameter("id");
 		String identifier = userBadge.getQueryParameter("identifier");
+		Timber.d("Scanning in user %s as %s", textId, identifier);
 		if (textId == null) {
 			Toast.makeText(getContext(), "Failed to scan any id", Toast.LENGTH_SHORT).show();
+			return;
 		}
 
 		int id = Integer.valueOf(textId);

